make rpi and make clean-rpi rpi both fail

sudo apt-get update
pi@raspberrypi ~ $ sudo apt-get install cmake g++-4.7 libcurl4-openssl-dev
Reading package lists... Done
Building dependency tree
Reading state information... Done
cmake is already the newest version.
g++-4.7 is already the newest version.
libcurl4-openssl-dev is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 38 not upgraded.
pi@raspberrypi ~ $ cd ~
pi@raspberrypi ~ $ git clone https://github.com/tangrams/tangram-es.git
fatal: destination path 'tangram-es' already exists and is not an empty directory.
pi@raspberrypi ~ $ git pull
fatal: Not a git repository (or any of the parent directories): .git
pi@raspberrypi ~ $ cd tangram-es
pi@raspberrypi ~/tangram-es $ git pull
Already up-to-date.
pi@raspberrypi ~/tangram-es $ git clean -n
pi@raspberrypi ~/tangram-es $ git submodule update
pi@raspberrypi ~/tangram-es $ git submodule init && git submodule update
Submodule 'core/include/libtess2/libtess2' () registered for path 'core/dependencies/tess2/libtess2'
Submodule 'core/include/fontstash-es' () registered for path 'core/include/fontstash-es'
Submodule 'core/include/glm' () registered for path 'core/include/glm'
pi@raspberrypi ~/tangram-es $ make clean-rpi rpi
rm -rf build/rpi
-- The C compiler identification is GNU 4.7.2
-- The CXX compiler identification is GNU 4.7.2
-- Check for working C compiler: /usr/bin/gcc
-- Check for working C compiler: /usr/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Platform chosen : RASPBERRYPI
-- Including raspberrypi dependent cmake file : /home/pi/tangram-es/toolchains/raspberrypi.cmake
-- Configuring done
-- Generating done
-- Build files have been written to: /home/pi/tangram-es/build/rpi
make[1]: Entering directory '/home/pi/tangram-es/build/rpi'
make[2]: Entering directory '/home/pi/tangram-es/build/rpi'
make[3]: Entering directory '/home/pi/tangram-es/build/rpi'
Scanning dependencies of target libtess2
make[3]: Leaving directory '/home/pi/tangram-es/build/rpi'
make[3]: Entering directory '/home/pi/tangram-es/build/rpi'
[  2%] Building C object core/dependencies/tess2/CMakeFiles/libtess2.dir/libtess2/Source/geom.c.o
[  4%] Building C object core/dependencies/tess2/CMakeFiles/libtess2.dir/libtess2/Source/tess.c.o
[  6%] Building C object core/dependencies/tess2/CMakeFiles/libtess2.dir/libtess2/Source/priorityq.c.o
[  9%] Building C object core/dependencies/tess2/CMakeFiles/libtess2.dir/libtess2/Source/sweep.c.o
[ 11%] Building C object core/dependencies/tess2/CMakeFiles/libtess2.dir/libtess2/Source/dict.c.o
[ 13%] Building C object core/dependencies/tess2/CMakeFiles/libtess2.dir/libtess2/Source/bucketalloc.c.o
[ 15%] Building C object core/dependencies/tess2/CMakeFiles/libtess2.dir/libtess2/Source/mesh.c.o
Linking C static library ../../../lib/liblibtess2.a
make[3]: Leaving directory '/home/pi/tangram-es/build/rpi'
[ 15%] Built target libtess2
make[3]: Entering directory '/home/pi/tangram-es/build/rpi'
Scanning dependencies of target core
make[3]: Leaving directory '/home/pi/tangram-es/build/rpi'
make[3]: Entering directory '/home/pi/tangram-es/build/rpi'
[ 18%] Building CXX object core/CMakeFiles/core.dir/src/text/textBuffer.cpp.o
[ 20%] Building CXX object core/CMakeFiles/core.dir/src/text/fontContext.cpp.o
[ 22%] Building CXX object core/CMakeFiles/core.dir/src/util/mapProjection.cpp.o
[ 25%] Building CXX object core/CMakeFiles/core.dir/src/util/texture.cpp.o
[ 27%] Building CXX object core/CMakeFiles/core.dir/src/util/builders.cpp.o
[ 29%] Building CXX object core/CMakeFiles/core.dir/src/util/vertexLayout.cpp.o
[ 31%] Building CXX object core/CMakeFiles/core.dir/src/util/geoJson.cpp.o
[ 34%] Building CXX object core/CMakeFiles/core.dir/src/util/rectangle.cpp.o
[ 36%] Building CXX object core/CMakeFiles/core.dir/src/util/error.cpp.o
[ 38%] Building CXX object core/CMakeFiles/core.dir/src/util/shaderProgram.cpp.o
[ 40%] Building CXX object core/CMakeFiles/core.dir/src/util/geom.cpp.o
[ 43%] Building CXX object core/CMakeFiles/core.dir/src/util/vboMesh.cpp.o
[ 45%] Building CXX object core/CMakeFiles/core.dir/src/util/pbfParser.cpp.o
[ 47%] Building CXX object core/CMakeFiles/core.dir/src/data/dataSource.cpp.o
[ 50%] Building CXX object core/CMakeFiles/core.dir/src/data/geoJsonSource.cpp.o
[ 52%] Building CXX object core/CMakeFiles/core.dir/src/data/protobufSource.cpp.o
[ 54%] Building CXX object core/CMakeFiles/core.dir/src/tile/labels/label.cpp.o
[ 56%] Building CXX object core/CMakeFiles/core.dir/src/tile/labels/labelContainer.cpp.o
[ 59%] Building CXX object core/CMakeFiles/core.dir/src/tile/mapTile.cpp.o
[ 61%] Building CXX object core/CMakeFiles/core.dir/src/tile/tileManager.cpp.o
[ 63%] Building CXX object core/CMakeFiles/core.dir/src/tile/tileWorker.cpp.o
[ 65%] Building CXX object core/CMakeFiles/core.dir/src/style/polylineStyle.cpp.o
[ 68%] Building CXX object core/CMakeFiles/core.dir/src/style/style.cpp.o
[ 70%] Building CXX object core/CMakeFiles/core.dir/src/style/debugStyle.cpp.o
[ 72%] Building CXX object core/CMakeFiles/core.dir/src/style/polygonStyle.cpp.o
[ 75%] Building CXX object core/CMakeFiles/core.dir/src/style/debugTextStyle.cpp.o
[ 77%] Building CXX object core/CMakeFiles/core.dir/src/style/fontStyle.cpp.o
[ 79%] Building CXX object core/CMakeFiles/core.dir/src/view/view.cpp.o
/home/pi/tangram-es/core/src/view/view.cpp: In function âvoid scanLine(int, int, int, int, std::set<TileID>&)â:
/home/pi/tangram-es/core/src/view/view.cpp:247:16: error: âclass std::set<TileID>â has no member named âemplaceâ
core/CMakeFiles/core.dir/build.make:675: recipe for target 'core/CMakeFiles/core.dir/src/view/view.cpp.o' failed
make[3]: *** [core/CMakeFiles/core.dir/src/view/view.cpp.o] Error 1
make[3]: Leaving directory '/home/pi/tangram-es/build/rpi'
CMakeFiles/Makefile2:112: recipe for target 'core/CMakeFiles/core.dir/all' failed
make[2]: *** [core/CMakeFiles/core.dir/all] Error 2
make[2]: Leaving directory '/home/pi/tangram-es/build/rpi'
Makefile:72: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/home/pi/tangram-es/build/rpi'
Makefile:160: recipe for target 'rpi' failed
make: *** [rpi] Error 2

the export statement made it go further to 97% bit it still crashed:
[ 97%] Building CXX object CMakeFiles/tangram.dir/rpi/src/main.cpp.o
/home/pi/tangram-es/rpi/src/main.cpp:49:2: warning: 'typedef' was ignored in this declaration [enabled by default]
 };
  ^
/home/pi/tangram-es/rpi/src/main.cpp: In function 'void initOpenGL()':
/home/pi/tangram-es/rpi/src/main.cpp:179:124: warning: invalid conversion from 'int' to 'DISPMANX_TRANSFORM_T' [-fpermissive]
                                                &src_rect, DISPMANX_PROTECTION_NONE, 0 /*alpha*/, 0/*clamp*/, 0/*transform*/);
                                                                                                                            ^
In file included from /opt/vc/include/bcm_host.h:50:0,
                 from /home/pi/tangram-es/core/src/gl.h:30,
                 from /home/pi/tangram-es/rpi/src/main.cpp:14:
/opt/vc/include/interface/vmcs_host/vc_dispmanx.h:85:44: warning:   initializing argument 10 of 'DISPMANX_ELEMENT_HANDLE_T vc_dispmanx_element_add(DISPMANX_UPDATE_HANDLE_T, DISPMANX_DISPLAY_HANDLE_T, int32_t, const VC_RECT_T*, DISPMANX_RESOURCE_HANDLE_T, const VC_RECT_T*, DISPMANX_PROTECTION_T, VC_DISPMANX_ALPHA_T*, DISPMANX_CLAMP_T*, DISPMANX_TRANSFORM_T)' [-fpermissive]
 VCHPRE_ DISPMANX_ELEMENT_HANDLE_T VCHPOST_ vc_dispmanx_element_add ( DISPMANX_UPDATE_HANDLE_T update, DISPMANX_DISPLAY_HANDLE_T display,
                                            ^
/home/pi/tangram-es/rpi/src/main.cpp:232:70: error: cannot allocate an object of abstract type 'VboMesh'
         mouseMesh = std::shared_ptr<VboMesh>(new VboMesh(vertexLayout));
                                                                      ^
In file included from /home/pi/tangram-es/rpi/src/main.cpp:18:0:
/home/pi/tangram-es/core/src/util/vboMesh.h:16:7: note:   because the following virtual functions are pure within 'VboMesh':
 class VboMesh {
       ^
/home/pi/tangram-es/core/src/util/vboMesh.h:50:18: note:        virtual void VboMesh::compileVertexBuffer()
     virtual void compileVertexBuffer() = 0;
                  ^
/home/pi/tangram-es/rpi/src/main.cpp:233:20: error: 'class VboMesh' has no member named 'addVertices'
         mouseMesh->addVertices((GLbyte*)vertices.data(), vertices.size());
                    ^
/home/pi/tangram-es/rpi/src/main.cpp:234:20: error: 'class VboMesh' has no member named 'addIndices'
         mouseMesh->addIndices(indices.data(), indices.size());
                    ^
CMakeFiles/tangram.dir/build.make:54: recipe for target 'CMakeFiles/tangram.dir/rpi/src/main.cpp.o' failed
make[3]: *** [CMakeFiles/tangram.dir/rpi/src/main.cpp.o] Error 1
make[3]: Leaving directory '/home/pi/tangram-es/build/rpi'
CMakeFiles/Makefile2:60: recipe for target 'CMakeFiles/tangram.dir/all' failed
make[2]: *** [CMakeFiles/tangram.dir/all] Error 2
make[2]: Leaving directory '/home/pi/tangram-es/build/rpi'
Makefile:72: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/home/pi/tangram-es/build/rpi'
Makefile:160: recipe for target 'rpi' failed
make: *** [rpi] Error 2

I tried again now theres a different problem at 100%
[100%] Building CXX object CMakeFiles/tangram.dir/rpi/src/platform_rpi.cpp.o
/home/pi/tangram-es/rpi/src/platform_rpi.cpp: In function 'bool isContinuousRendering()':
/home/pi/tangram-es/rpi/src/platform_rpi.cpp:26:1: warning: no return statement in function returning non-void [-Wreturn-type]
 }
 ^
Linking CXX executable bin/tangram
/usr/bin/ld: lib/libcore.a(tileWorker.cpp.o): undefined reference to symbol 'pthread_create@@GLIBC_2.4'
//lib/arm-linux-gnueabihf/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
CMakeFiles/tangram.dir/build.make:112: recipe for target 'bin/tangram' failed
make[3]: *** [bin/tangram] Error 1
make[3]: Leaving directory '/home/pi/tangram-es/build/rpi'
CMakeFiles/Makefile2:60: recipe for target 'CMakeFiles/tangram.dir/all' failed
make[2]: *** [CMakeFiles/tangram.dir/all] Error 2
make[2]: Leaving directory '/home/pi/tangram-es/build/rpi'
Makefile:72: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/home/pi/tangram-es/build/rpi'
Makefile:160: recipe for target 'rpi' failed
make: *** [rpi] Error 2

--- a/rpi/src/main.cpp
+++ b/rpi/src/main.cpp
@@ -15,7 +15,7 @@
 
 #include "util/shaderProgram.h"
 #include "util/vertexLayout.h"
-#include "util/vboMesh.h"
+#include "util/typedMesh.h"
 #include "util/geom.h"
 
 #define KEY_ESC      113    // q
@@ -102,7 +102,8 @@ struct PosUVColorVertex {
     // Color Data
     GLuint abgr;
 };
-std::shared_ptr<VboMesh> mouseMesh;
+typedef TypedMesh<PosUVColorVertex> Mesh;
+std::shared_ptr<Mesh> mouseMesh;
 
 static void initOpenGL(){
     bcm_host_init();
@@ -229,9 +230,9 @@ static void initOpenGL(){
             indices.push_back(2); indices.push_back(3); indices.push_back(0);
         }
         
-        mouseMesh = std::shared_ptr<VboMesh>(new VboMesh(vertexLayout));
-        mouseMesh->addVertices((GLbyte*)vertices.data(), vertices.size());
-        mouseMesh->addIndices(indices.data(), indices.size());
+        mouseMesh = std::shared_ptr<Mesh>(new Mesh(vertexLayout, GL_TRIANGLES));
+        mouseMesh->addVertices(std::move(vertices), std::move(indices));
+        mouseMesh->compileVertexBuffer();
     }
 }
 

--- a/toolchains/raspberrypi.cmake
+++ b/toolchains/raspberrypi.cmake
@@ -1,6 +1,6 @@
 # options
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -fpermissive -g")
-set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_CXX_FLAGS} -L/opt/vc/lib/ -lGLESv2 -lEGL -lbcm_host -lvchiq_arm -lvcos -lrt")
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_CXX_FLAGS} -L/opt/vc/lib/ -lGLESv2 -lEGL -lbcm_host -lvchiq_arm -lvcos -lrt -lpthread")
 set(CXX_FLAGS_DEBUG "-g -O0")
 set(EXECUTABLE_NAME "tangram")
 

